### PR TITLE
Qualtrics SDK

### PR DESCRIPTION
**Pull request by Adamwgoh**
Qualtrics SDK connector.

### CODE REVIEW

**Comment by Adamwgoh** [comment]
@dkogias : @amland has a client requesting to use Qualtrics in Prod, what would be the process be for me to test it and ready for production?

**Comment by amland** [comment]
Hi Dimitris,

Sullivan Solar wants to use this right away. They want a response by EOD if
we can do this. They are quickbase, infusionsoft customer.



Let Workato recipes do the work for you!
+1 (408) 316 8989 | www.workato.com | LinkedIn
<http://linkedin.com/pub/amlan-debnath/2/55/676/en>

**Comment by dkogias** [comment]
@Adamwgoh @amland Is this needed as a "global" adapter?  It should work as is, I'm assuming the attached SDK adapter already works.

**Comment by Adamwgoh** [comment]
@dkogias Yes it works. However, should I be creating test_cases using rspec/vcr on this? Eeshan mentioned to me about it before.

File: api_key_auth/qualtrics.rb [hunk 1]
```diff
@@ -0,0 +1,188 @@
+{
+  title: 'Qualtrics',
+
+  connection: {
+    fields: [
+      { name: 'host', control_type: 'subdomain', url: '.qualtrics.com', optional: false },
+      { name: 'username', label: 'username/email', optional: false },
+      { name: 'api_key', control_type: 'password', optional: false },
```
Comment (eeshan92) [inline api_key_auth/qualtrics.rb]: I believe it's labelled **token** in the UI. Let's change to avoid confusion.

File: api_key_auth/qualtrics.rb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      },
+      
+      output_fields: ->(object_definitions){
+        [
+          { name: "result", type: :array, of: :object, properties: object_definitions['response_detail'] }
+        ]
+      }
+    },
+    
+    download_responses: {
```
Comment (eeshan92) [inline api_key_auth/qualtrics.rb]: Do we support this? (CSV format) If not, it's best to remove it.

File: api_key_auth/qualtrics.rb [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      input_fields: ->() {},
+
+      webhook_subscribe: ->(webhook_url,connection,input,recipe_id) {
+        post("https://#{connection['host']}.qualtrics.com/API/v3/eventsubscriptions").
+          payload(publicationUrl: webhook_url,
+                  topics: 'surveyengine.completedResponse.'+ connection['survey_id'])
+      },
+
+      
+      webhook_unsubscribe: ->(webhook, connection) {
```
Comment (eeshan92) [inline api_key_auth/qualtrics.rb]: There is no `connection` parameter atm

**Pull request closed without merging**